packages/react-flex/src/components/Flex/useFlexStyles.ts

```diff
@@ -16,11 +25,22 @@ export const useFlexStyles = (state: FlexState): FlexState => {
   const styles = useStyles();
   state.className = mergeClasses(styles.root, state.className);
 
-  // TODO: Hook up slot styles, for example:
-  // const exampleSlotClassName = useExampleSlotStyles(state);
-  // if (state.exampleSlot) {
-  //   state.exampleSlot.className = mergeClasses(exampleSlotClassName, state.exampleSlot.className);
-  // }
+  const direction = state.direction ?? 'row';
+  const horizontalAlign = state.horizontalAlign ?? 'normal';
+  const verticalAlign = state.verticalAlign ?? 'normal';
+
+  state.style = {
+    ...state.style,
+    display: state.inline ? 'inline-flex' : 'flex',
+    flexDirection: direction,
+    flexWrap: state.wrap ? 'wrap' : 'nowrap',
+    justifyContent: direction.startsWith('row') ? horizontalAlign : verticalAlign,
+    alignItems: direction.startsWith('row') ? verticalAlign : horizontalAlign,
+
+    ['--grow' as string]: state.grow === undefined ? 0 : state.grow,
+    ['--shrink' as string]: state.shrink === undefined ? 1 : state.shrink,
+    ['--gap' as string]: state.gap === undefined ? 0 : state.gap,
```

Comment:
@layershifter  do you have any better ideas for this with `makeStyles` ? having inline styles will be very problematic for overrides, the only way to override would be with `important` AFAIK